src/kbmod/analysis_utils.py

```diff
@@ -931,105 +758,82 @@ def apply_stamp_filter(
 
         # Save some useful helper data.
         num_times = search.get_num_images()
-        num_indices = len(keep["lc_index"])
-        keep["final_results"] = np.array([])
-        keep["stamps"] = []
         all_valid_inds = []
-        keep2 = []
 
         # Run the stamp creation and filtering in batches of chunk_size.
         print("---------------------------------------")
         print("Applying Stamp Filtering")
         print("---------------------------------------", flush=True)
         start_time = time.time()
         start_idx = 0
-        passing_stamps_idx = []
-        num_results = len(keep["results"])
-        if num_results > 0:
-            print("Stamp filtering %i results" % num_results)
-            while start_idx < num_results:
-                end_idx = min([start_idx + chunk_size, num_results])
+        if result_list.num_results() > 0:
+            print("Stamp filtering %i results" % result_list.num_results())
+            while start_idx < result_list.num_results():
+                end_idx = min([start_idx + chunk_size, result_list.num_results()])
 
                 # Create a subslice of the results and the Boolean indices (as TrajectoryResults).
                 # Note that the sum stamp type does not filter out lc_index.
+                inds_to_use = [i for i in range(start_idx, end_idx)]
                 results_slice = []
-                for i in range(start_idx, end_idx):
-                    if i < num_indices and params.stamp_type != kb.StampType.STAMP_SUM:
-                        trj = kb.trj_result(keep["results"][i], num_times, keep["lc_index"][i])
-                    else:
-                        trj = kb.trj_result(keep["results"][i], num_times)
-                    results_slice.append(trj)
+                if params.stamp_type != kb.StampType.STAMP_SUM:
+                    results_slice = result_list.trj_result_list(indices_to_use=inds_to_use)
+                else:
+                    trj_list = result_list.trajectory_list(indices_to_use=inds_to_use)
+                    results_slice = [kb.trj_result(x, num_times) for x in trj_list]
 
                 # Create and filter the results.
                 stamps_slice = search.gpu_coadded_stamps(results_slice, params)
                 for ind, stamp in enumerate(stamps_slice):
                     if stamp.get_width() > 1:
-                        keep["stamps"].append(np.array(stamp))
+                        result_list.results[ind + start_idx].stamp = np.array(stamp)
                         all_valid_inds.append(ind + start_idx)
 
                 # Move to the next chunk.
                 start_idx += chunk_size
+                del stamps_slice
```

Comment:
Is this required?

Reply:
I don't think so. It was in the original code, so I just moved it into the loop. But I don't think this is actually needed, so let me remove it.